Profile controller

Greengrer opened the pull request.

TEdwardJ committed:
Profile controller moved to rest:
 - change discogs user`s name functionality;
 - change password functionality;
 - UserDao has got 2 new methods for password changing and for discogsUserName editing;
  - UserService has got 2 new methods for password changing and for discogsUserName editing;

Greengrer committed:
Merge branch 'profile_controller' of https://github.com/littlepony99/wax-deals into profile_controller

Greengrer committed:
Changed profile controller test to use @beforeEach.

TEdwardJ committed:
Profile controller moved to rest:
 - Order of parameters in the DefaultUserService.changeDiscogsUserName() method in order to make the order the same as in JdbcUserDao.changeDiscogsUserName();

TEdwardJ committed:
Profile controller:
 - profile management logic moved to ProfileManagementService from ProfileController;
 - JwtTokenProvider moved to impl package;
 - ProfileController test fix;

TEdwardJ committed:
Change Password functionality:
 - controller should receive old password along with new password and new password confirmation;

TEdwardJ committed:
Password Recovery functionality:
 - rest-styled "/password-recovery/{recoveryToken}" endpoint added to the password recovery controller;;

TEdwardJ committed:
Password Recovery test:
 - elimination of unnecessary code;

TEdwardJ committed:
Email Confirmation new endpoint:
 - confirmation token is expected to be sent in PUT request body;

TEdwardJ committed:
Possibility to edit user`s email:
 - Besides of possibility to edit the discogsUserName attribute, possibility to edit user`s email added;

TEdwardJ committed:
OneVinylOfferDto Mapping changes:
 - id field is added into OneVinylOfferDto;

TEdwardJ committed:
OneVinylOfferDto Mapping changes:
 - id field mapping issue fix - ambiguity for id fields eliminated;

TEdwardJ committed:
Profile Management changes:
 - Http Code 400 is returned in case the user specified wrong old password during password change process;
 - new JWT token is returned in case the user changed their email;

Greengrer commented on src/main/java/com/vinylteam/vinyl/security/SecurityConfiguration.java:
```diff
@@ -16,12 +16,10 @@
 import org.springframework.web.servlet.config.annotation.CorsRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
-import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+
```
extra lines should be removed

Greengrer reviewed (commented):
Just a couple exatra lines in one of the files so far

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -100,4 +108,21 @@ public void setUserStatusTrue(long id) {
+                .addValue("new_email", email)
+                .addValue("discogs_user_name", discogsUserName)
+                .addValue("email", user.getEmail());
+        namedParameterJdbcTemplate.update(UPDATE_PROFILE_FIELDS, sqlParameterSource);
```
please remove variable declaration

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java:
```diff
@@ -100,4 +108,21 @@ public void setUserStatusTrue(long id) {
+        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource()
+                .addValue("email", user.getEmail())
+                .addValue("password", user.getPassword());
+        namedParameterJdbcTemplate.update(UPDATE_PASSWORD, sqlParameterSource);
```
please remove variable declaration

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/JwtAuthenticationException.java:
```diff
@@ -0,0 +1,13 @@
+import org.springframework.security.core.AuthenticationException;
+
+public class JwtAuthenticationException extends AuthenticationException {
+    public JwtAuthenticationException(String msg, Throwable t) {
```
bad parameters names

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/JwtAuthenticationException.java:
```diff
@@ -0,0 +1,13 @@
+    }
+
+    public JwtAuthenticationException(String msg) {
+        super(msg);
```
bad parameter name

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java:
```diff
@@ -120,6 +118,42 @@ public User findByEmail(String email) {
+        boolean checkOldPassword = securityService.validateIfPasswordMatches(user, oldPassword.toCharArray());
+        if (!checkOldPassword) {
+            throw new BadCredentialsException(UserErrors.WRONG_PASSWORD_ERROR.getMessage());
+            //throw new RuntimeException(UserErrors.WRONG_PASSWORD_ERROR.getMessage());
```
YAGNI

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/JwtTokenProvider.java:
```diff
@@ -81,7 +84,8 @@ public boolean isTokenValid(String token) {
         } catch (JwtException | IllegalArgumentException e) {
             log.error("JWT token is expired or invalid", e);
-            return false;
+            //return false;
```
YAGNI

TEdwardJ committed:
Profile Controller PR review:
 - elimination of commented code;
 - changes in the variable naming;
 - elimination of redundant empty lines;;

TEdwardJ committed:
Profile Controller PR review:
 - JwtAuthenticationException constructor Throwable-typed parameter has been renamed;
